### PR TITLE
Fixes & Token deployment

### PR DESCRIPTION
**Pull request by rrecuero**

### CODE REVIEW

File: script/deploy.s.sol [hunk 1]
```diff
@@ -80,7 +80,7 @@ contract DeployerScript is Create2Helper, ArtifactsReader {
     // @dev this is used for tests
     function runAndReturnResults(uint256 _privateKey) public returns (DeployedContracts memory contracts) {
         privateKey = _privateKey;
-        write = true;
+        write = false;
```
Comment (fedealconada) [inline script/deploy.s.sol:83]: not sure why I changed this, but seems to be running fine! :)

File: test/helpers/ArtifactsReader.sol [hunk 2]
```diff
@@ -20,4 +20,17 @@ abstract contract ArtifactsReader is Script {
             return address(0);
         }
     }
+
+    function _getOtherChainDeployment(string memory contractName, uint256 chainId) internal virtual returns (address) {
```
Comment (fedealconada) [inline test/helpers/ArtifactsReader.sol:24]:
```suggestion
    function _getChainDeployment(string memory contractName, uint256 chainId) internal virtual returns (address) {
```

you can just keep the same method name as the one above and make the one above to call this one, would look something like this:

```
function _getAddressesFile() internal view virtual returns (string memory) {
        return _getAddressesFile(block.chainid);
    }

    function _getAddressesFile(uint256 chainId) internal view virtual returns (string memory) {
        string memory root = vm.projectRoot();
        return string.concat(root, "/test/artifacts/", vm.toString(chainId), "/addresses.json");
    }

    function _getChainDeployment(string memory contractName) internal virtual returns (address) {
        return _getChainDeployment(contractName, block.chainid);
    }

    function _getChainDeployment(string memory contractName, uint256 chainId) internal virtual returns (address) {
        try vm.readFile(_getAddressesFile(chainId)) returns (string memory json) {
            try vm.parseJsonAddress(json, string.concat(".", contractName)) returns (address addr) {
                return addr;
            } catch {
                return address(0);
            }
        } catch {
            return address(0);
        }
    }
    ```

File: test/SharedSetup.t.sol [hunk 3]
```diff
@@ -97,8 +97,9 @@ contract SharedSetup is UserOp, AATestScaffolding, ArtifactsReader {
             changeWalletOwner(_owner, _kycProvider);
         } else {
             console.log("Running tests locally");
+            console.log("Chain ID", 99999999999999);
+            vm.chainId(99999999999999);
```
Comment (fedealconada) [inline test/SharedSetup.t.sol:101]: i think we don't need this, right?

**Review by fedealconada — APPROVED** [review]
lgtm, only a few small comments

File: test/helpers/ArtifactsReader.sol (hunk 2 above)
Comment (rrecuero) [reply test/helpers/ArtifactsReader.sol]: done

**Pull request merged**